common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -739,6 +792,40 @@ public boolean isTerminated() {
         return state == ST_TERMINATED;
     }
 
+    @Override
+    public boolean isSuspended() {
+        return state == ST_SUSPENDED || state == ST_SUSPENDING;
```

Comment:
snapshot `state` just once or we risk to capture a change while checking again

Reply:
Good catch!